### PR TITLE
[#19] Replace nba endpoint

### PR DESCRIPTION
**Pull request by kammanz**
Completed issue 19. There's a problem though, it seems there is only one size for images and we wanted a large image to appear on the right side when user selects player. It seems there are multiples sizes for video, but not image, according to the bottom of the npm page ...
https://www.npmjs.com/package/nba-api-client  

If I missed something please let me know.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "name": "my-app",
   "version": "0.1.0",
   "private": true,
+  "proxy": "http://data.nba.net",
```
Comment (Tumne) [inline package.json]: If not using, remove

File: src/actions/actions.js [hunk 2]
```diff
@@ -1,19 +1,15 @@
...
         return null;
-    }
-    return {
-        type: 'SELECT_PLAYER',
-        payload: player
-    }
+    };
+
+    const response = await dataNbaNet.get("/prod/v1/2019/players/" + playerId + "_profile.json");
+    dispatch({ type: 'SELECT_PLAYER', payload: {...response.data.league.standard.stats.latest, playerName, playerId }});
```
Comment (Tumne) [inline src/actions/actions.js:14]: Missing space before curly bracket `{ ...response.data`

File: src/components/PlayerDetails.js [hunk 3]
```diff
@@ -1,21 +1,28 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
-const PlayerDetails = ({ player }) => {
-    if (!player) {
+const PlayerDetails = (props)=> {
+        const { selectPlayer } = props;
```
Comment (Tumne) [inline src/components/PlayerDetails.js]: Tabbed over too much

File: src/components/PlayerDetails.js [hunk 4]
```diff
@@ -1,21 +1,28 @@
...
-            <div>Field Goal Percentage:{player.FG_PCT}</div>
-            <div>Points: {player.PTS}</div>
-            <div>Rebounds: {player.REB}</div>
+            <h3>Stats</h3>
+            <h4>{selectPlayer.playerName}</h4>
+            <img src={`https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/1610612761/2019/260x190/${selectPlayer.playerId}.png`} />            
+            <div>Total Points: {selectPlayer.points}</div>
+            <div>Points per Game: {selectPlayer.ppg}</div>
+            <div>Blocks: {selectPlayer.blocks}</div>
+            <div>Steals: {selectPlayer.steals}</div>
```
Comment (Tumne) [inline src/components/PlayerDetails.js]: Can we use a destruct for all these values

`const { playerName, playerId, points, ppg, blocks, steals } = selectPlayer;`

File: src/components/Players.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+import { getPlayerDetails } from '../actions/actions.js';
+
+class Players extends React.Component {
+    componentDidMount() {
+
+        this.props.getPlayers();
+        // this.props.getPlayerDetails();
+        console.log('this.props.getPlayers()', this.props.getPlayers());
+        // console.log('this.props.getPlayers()', this.props.getPlayerDetails());
+        // this.props.playerHeadshot();
```
Comment (Tumne) [inline src/components/Players.js]: Remove commented out code.

File: src/components/Players.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import { selectPlayer } from '../actions/actions.js';
+import { getPlayerHeadshot } from '../actions/actions.js';
+import { getPlayerDetails } from '../actions/actions.js';
+
+class Players extends React.Component {
+    componentDidMount() {
+
+        this.props.getPlayers();
+        // this.props.getPlayerDetails();
+        console.log('this.props.getPlayers()', this.props.getPlayers());
```
Comment (Tumne) [inline src/components/Players.js]: Always remove console.log in PR. Unless, you need to leave a console, in those very few cases, use `console.info('this.props.getPlayers()', this.props.getPlayers());`

I'll let you decide between one of these choices

File: src/components/Players.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        // console.log('this.props.getPlayers()', this.props.getPlayerDetails());
+        // this.props.playerHeadshot();
+      
+    }
+
+    renderPlayers() {
+        const { players } = this.props;
+        console.log('players', players);
```
Comment (Tumne) [inline src/components/Players.js]: Same comment as above, remove

File: src/components/Players.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        //     return (
+        //         <div key={index} onClick={() => this.props.selectPlayer(player)}>
+        //             <h4>{player.PLAYER}</h4>
+        //             <p>Height: {player.HEIGHT}'</p>
+        //             <p>Weight: {player.WEIGHT} Lbs.</p>
+        //             <p>Position: {player.POSITION}</p>
+        //         </div>
+        //     ); 
+        // });
+    }
```
Comment (Tumne) [inline src/components/Players.js]: Remove this unused function

File: src/components/Players.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    //     //     return (
+    //     //         <div key={index} onClick={() => this.props.selectPlayer(player)}>
+    //     //             <h4>{player.PLAYER}</h4>
+    //     //             <p>Height: {player.HEIGHT}'</p>
+    //     //             <p>Weight: {player.WEIGHT} Lbs.</p>
+    //     //             <p>Position: {player.POSITION}</p>
+    //     //         </div>
+    //     //     ); 
+    //     // });
+    // }
```
Comment (Tumne) [inline src/components/Players.js]: Remove commented out code

File: src/components/Players.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+    render() {
+        console.log(this.props.players, 'this.props render in playerslist' );
```
Comment (Tumne) [inline src/components/Players.js]: Remove this

File: src/components/Players.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+                <h3>Players</h3>
+                <div>{this.renderPlayers()}</div>
+                <br/>
+            </div>
+        )
+    }
+}
+
+const mapStateToProps = (state) => {
+    console.log(state.players);
```
Comment (Tumne) [inline src/components/Players.js]: Remove

File: src/components/Players.js [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+    return { players: state.players };
+    
+    // return { playerHeadshot: state.playerHeadshot }
```
Comment (Tumne) [inline src/components/Players.js]: Remove

File: src/components/Players.js [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    // }
+
+    render() {
+        console.log(this.props.players, 'this.props render in playerslist' );
+
+        return (
+            <div>
+                <h3>Players</h3>
+                <div>{this.renderPlayers()}</div>
+                <br/>
```
Comment (Tumne) [inline src/components/Players.js]: Remove this `<br/>` try to avoid these in general. We should be using css instead.

**Review by Tumne — CHANGES_REQUESTED** [review]
Works well. There are some clean up that's needed. 

Also I noticed that this app is not properly organized and would be good to restructure the codebase before it starts to grow. I will create a separate ticket for this.

File: src/components/Players.js (hunk 5 above)
Comment (kammanz) [reply src/components/Players.js]: I deleted Players.js altogether, extraneous file, PlayersList.js is the one being used.

File: src/actions/actions.js (hunk 2 above)
Comment (Tumne) [reply src/actions/actions.js:14]: Not competed yet

**Review by Tumne — APPROVED** [review]

**Pull request merged**